### ISSUE TITLE
hildon-desktop(-light): without animations

### ISSUE THREAD

**Issue by MerlijnWajer** [maemo-leste/bugtracker#51]
Let's package a hildon-deskto-light package where most of the GPU intensive shaders and other animations are ripped out, for platforms where we only have llvmpipe for now.

**Comment by MerlijnWajer** [comment]
< Wizzup> maybe we can make hildon-desktop-light that uses no shaders ;)
< freemangordon> yes, that'd be easy
< freemangordon> shaders are actually ClutterEffect, by just not adding it to the actor you're not applying shaders

This, combined with very short times for animations might make a real difference

**Comment by pavelmachek** [comment]
Is there any progress here? Desktop is slow even when typing in the terminal, which is quite annoying. Are shaders also responsible for that?

Is there way to run different window manager? Droid 4 has extra buttons at right side of screen, perhaps those could be used for switching/closing windows until better solution is available.

**Comment by dderby** [comment]
@pavelmachek, I've been using a couple of methods to get an on-device terminal outside of Hildon.

One method is to install the `kbd` package.  Then you can change the foreground terminal with the `chvt` command with Hildon still running.

The other method is to prevent Hildon from starting.  The simplest way to do this right now is with this hack:
```
# mv /etc/init.d/dsme /etc/init.d/dsme_
```
With Hildon out of the way you can install `xinit` and install and use any window manager you like.  Right now, you can't just run `startx` on its own with no arguments as there is a Hildon Xsession script which breaks starting the default window manager set by `/usr/bin/x-window-manager`.  If you specify the executable, for example `startx /usr/bin/i3` to start i3 wm, it should work.

**Comment by MerlijnWajer** [comment]
Yes, even typing in terminal is quite slow. The best solution is to get 3d to work on the droid, but that's for later.

I believe that the shaders are not responsible for this, but rather the compositing, or just the fact that everything is being drawn with opengl. I'd need to dive in a bit more to see if this can be done more efficiently. But ultimately, we'll want powervr to work on the droid. I suppose it shouldn't be -that- hard.

**Comment by pavelmachek** [comment]
@dderby:

Thanks a lot. Disabling dsme_ and then startx /usr/bin/something
... works for me... when using xfce4. I could not figure out how to
start mate session that way.

I was able to bind up/down/power key to home app/prev window/close
window, and that works well enough.

I thought that touch area on droid 4 is actually buttons ... it is
not. It seems to be touchscreen that extends beyond visible
screen. Fun. It would be quite nice to figure a way to use it in X.

-- 
(english) http://www.livejournal.com/~pavelmachek
(cesky, pictures) http://atrey.karlin.mff.cuni.cz/~pavel/picture/horses/blog.html
